### PR TITLE
Restructure how sheets are accessed in the object oriented structure of the SheetCollector - Issue #41

### PR DESCRIPTION
**Pull request by Makell-Logan**

### CODE REVIEW

**Comment by noorbuchi** [comment]
Please make sure that the build is passing. It looks like there are few issues where a variable is being used even though it's not defined.

File: sheetshuttle/sheet_collector.py [hunk 1]
```diff
@@ -286,12 +294,19 @@ def get_region(self, region_name: str):
...
+        requested_tab: Tab = self.tabs[sheet[region_name]]
+        print(requested_tab)
+        return requested_tab
 
     def print_sheet(self):
         """Iterate through self.regions and print the contents."""
-        for region_id, region in self.regions.items():
+        # update this with the new indexing structure
+        # Fixed indexing structure in this function
+        for region_id, region in self.tabs[sheet[regions.items()]]:
```
Comment (noorbuchi) [inline sheetshuttle/sheet_collector.py:309]: what is `sheet` referring to here? This variable doesn't look to be defined in this scope

**Review by noorbuchi — CHANGES_REQUESTED** [review]

**Comment by EstebanMendez01** [comment]
New changes done for the enhancement to pass all the checks. All the tests are passing now. Please review @noorbuchi @jjumadinova

**Review by EstebanMendez01 — DISMISSED** [review]

**Review by Danniyb — DISMISSED** [review]

**Review by tmiller-10 — DISMISSED** [review]

File: sheetshuttle/sheet_collector.py [hunk 2]
```diff
@@ -222,6 +222,12 @@ def authenticate_api(key_file):
         return credentials, service, sheets
 
 
+# Fix the failing tests due to the change in the indexing structure
+# Run: poetry run task test
+# go through through existing documentation and update the examples with the new indexing structure
+# Viewed docs
+# Update the ee_grade plugin with the new indexing structure
+# Changed some stuff in the plugin
```
Comment (noorbuchi) [inline sheetshuttle/sheet_collector.py:230]: Let's get rid of those unnecessary comments

File: sheetshuttle/sheet_collector.py [hunk 3]
```diff
@@ -275,9 +282,11 @@ def collect_regions(self):
...
                     data,
                 )
-                self.regions[region_object.full_name] = region_object
+                regions_dict[region_object.region_name] = region_object
+            self.tabs[sheet["name"]] = regions_dict
 
-    def get_region(self, region_name: str):
+    def get_tab(self, tab_name: str):
+        # change doc
         """Return a region object from the regions dictionary.
```
Comment (noorbuchi) [inline sheetshuttle/sheet_collector.py:290]: The documentation here needs to be revised to reflect what's being done

File: sheetshuttle/sheet_collector.py [hunk 4]
```diff
@@ -286,14 +295,18 @@ def get_region(self, region_name: str):
         Returns:
             Region: the region object from the self.regions dictionary
         """
-        requested_region: Region = self.regions[region_name]
-        return requested_region
+        # update this with the new indexing structure
+        # requested_region: Region = self.regions[region_name]
+        # return requested_region
+        # New Approach
```
Comment (noorbuchi) [inline sheetshuttle/sheet_collector.py:301]: Let's not keep these comments here

File: sheetshuttle/sheet_collector.py [hunk 5]
```diff
@@ -417,6 +430,8 @@ def region_to_pickle(self, directory: pathlib.PosixPath):
             directory (pathlib.PosixPath): path to the directory where the file
                 be stored
         """
+        # Make a decision on the new file naming style
+        # Fixed naming
```
Comment (noorbuchi) [inline sheetshuttle/sheet_collector.py:434]: Do we need these comments?

File: sheetshuttle/sheet_collector.py [hunk 6]
```diff
@@ -429,9 +444,11 @@ def region_to_json(self, directory: pathlib.PosixPath):
             directory (pathlib.PosixPath): path to the directory where the file
                 be stored
         """
+        # Make a decision on the new file naming style
```
Comment (noorbuchi) [inline sheetshuttle/sheet_collector.py:447]: Do we need these comments?

File: sheetshuttle/sheet_collector.py [hunk 7]
```diff
@@ -429,9 +444,11 @@ def region_to_json(self, directory: pathlib.PosixPath):
...
         self_data = {
             "region_name": self.region_name,
             "parent_name": self.parent_sheet_name,
+            # keep full_name, but use it just for opening the file
```
Comment (noorbuchi) [inline sheetshuttle/sheet_collector.py:451]: Do we need these comments?

File: sheetshuttle/sheet_collector.py [hunk 8]
```diff
@@ -443,3 +460,36 @@ def region_to_json(self, directory: pathlib.PosixPath):
...
+        Args:
+            region_name (str): name of the region to get
+
+        Returns:
+            Region: the region object from the self.regions dictionary
+        """
+        # update this with the new indexing structure
+        # requested_region: Region = self.regions[region_name]
+        # return requested_region
+        # New Approach
```
Comment (noorbuchi) [inline sheetshuttle/sheet_collector.py:493]: Do we need these comments?

**Review by noorbuchi — CHANGES_REQUESTED** [review]

**Comment by EstebanMendez01** [comment]
Fixed comments and documentation errors

**Review by noorbuchi — DISMISSED** [review]

**Review by burgess01 — DISMISSED** [review]
This looks great- good job!

**Review by noorbuchi — APPROVED** [review]

**Review by jjumadinova — APPROVED** [review]
Looks good now. Thank you!

**Pull request merged**